src/lib/components/Cytoscape.react.js

```diff
@@ -196,6 +200,81 @@ class Cytoscape extends Component {
         }, SELECT_THRESHOLD);
 
         // /////////////////////////////////////// EVENTS //////////////////////////////////////////
+        const updateTapContextMenu = (newContext) => {
+            this.props.setProps({ tapContextMenu: newContext });
+        }
+        const updateElements = (newElement) => {
+            var updatedElements = this.props.elements
+            updatedElements.push(newElement)
+            this.props.setProps({ elements: updatedElements });
+        }
+        const contextMenuDefaultFunctions = {
+            remove: function (event) {
+                var target = event.target || event.cyTarget;
+                target.remove();
+            },
+            add_node: function (event) {
+                var data = {
+                    group: 'nodes'
+                };
+                var pos = event.position || event.cyPosition;
+                cy.add({
+                    data: data,
+                    position: {
+                        x: pos.x,
+                        y: pos.y
+                    }
+                });
+            },
+            add_edge: function (event) {
+                const selectedNodeIds = selectedNodes.map(node => node.id());
+                if (selectedNodes.length === 2) {
+                    var newEdge = {
+                        data: {
+                            id: uuidv4(),
+                            group: 'edges',
+                            source: selectedNodeIds[0],
+                            target: selectedNodeIds[1],
+                        }
+                    }
+                    cy.add(newEdge)
+                    updateElements(newEdge)
+
+                    // this.props.setProps({ elements: newEdge })
+                };
+            },
+        }
+
+        const createMenuItems = (ctxMenuData) => {
+            var new_menu_items = []
+            for (const item of ctxMenuData) {
+                // use default javascript function
+                if (contextMenuDefaultFunctions.hasOwnProperty(item.onClickFunction)) {
+                    var onClickFunction = contextMenuDefaultFunctions[item.onClickFunction]
```

Comment:
Let's try to make full use of `let` and `const`, and avoid `var` entirely.

`new_menu_items` can be a `const`.
`onClickFunction` should be a `let` at the top of the for loop.